src/modules/70-pipeline/components/PipelineStudio/StepUtil.ts

```diff
@@ -418,78 +418,97 @@ export const validatePipeline = ({
   getString,
   path
 }: ValidatePipelineProps): FormikErrors<PipelineInfoConfig> => {
-  const errors = validateCICodebase({
-    pipeline,
-    template,
-    originalPipeline,
-    viewType,
-    getString,
-    path
-  })
-
-  if (getMultiTypeFromValue(template?.timeout) === MultiTypeInputType.RUNTIME) {
-    let timeoutSchema = getDurationValidationSchema({ minimum: '10s' })
-    if (viewType === StepViewType.DeploymentForm) {
-      timeoutSchema = timeoutSchema.required(getString?.('validation.timeout10SecMinimum'))
+  if (template.template) {
+    const errors = validatePipeline({
+      pipeline: pipeline.template?.templateInputs as PipelineInfoConfig,
+      template: template.template?.templateInputs as PipelineInfoConfig,
+      viewType,
+      originalPipeline: originalPipeline?.template?.templateInputs as PipelineInfoConfig,
+      getString
+    })
+    if (!isEmpty(errors)) {
+      return set({}, 'template.templateInputs', errors)
+    } else {
+      return {}
     }
-    const timeout = Yup.object().shape({
-      timeout: timeoutSchema
+  } else {
+    const errors = validateCICodebase({
+      pipeline,
+      template,
+      originalPipeline,
+      viewType,
+      getString,
+      path
     })
 
-    try {
-      timeout.validateSync(pipeline)
-    } catch (e) {
-      /* istanbul ignore else */
-      if (e instanceof Yup.ValidationError) {
-        const err = yupToFormErrors(e)
+    if (getMultiTypeFromValue(template?.timeout) === MultiTypeInputType.RUNTIME) {
+      let timeoutSchema = getDurationValidationSchema({ minimum: '10s' })
+      if (viewType === StepViewType.DeploymentForm) {
+        timeoutSchema = timeoutSchema.required(getString?.('validation.timeout10SecMinimum'))
+      }
+      const timeout = Yup.object().shape({
+        timeout: timeoutSchema
+      })
 
-        Object.assign(errors, err)
+      try {
+        timeout.validateSync(pipeline)
+      } catch (e) {
+        /* istanbul ignore else */
+        if (e instanceof Yup.ValidationError) {
+          const err = yupToFormErrors(e)
+
+          Object.assign(errors, err)
+        }
       }
     }
-  }
 
-  if (pipeline?.variables) {
-    const step = factory.getStep(StepType.CustomVariable)
-    const errorsResponse: any = step?.validateInputSet({ data: pipeline, template, getString, viewType })
+    if (pipeline?.variables) {
+      const step = factory.getStep(StepType.CustomVariable)
+      const errorsResponse: any = step?.validateInputSet({ data: pipeline, template, getString, viewType })
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
    const errorsResponse: unknown = step?.validateInputSet({ data: pipeline, template, getString, viewType })
```